src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_sysfs.c

```diff
@@ -279,20 +272,64 @@ kds_custat_raw_show(struct device *dev, struct device_attribute *attr, char *buf
 	ssize_t ret;
 
 	mutex_lock(&xdev->dev_lock);
-	ret = show_kds_custat_raw(&XDEV(xdev)->kds, buf);
+	ret = show_kds_custat_raw(&XDEV(xdev)->kds, buf, PAGE_SIZE);
 	mutex_unlock(&xdev->dev_lock);
 	return ret;
 }
 static DEVICE_ATTR_RO(kds_custat_raw);
 
+static ssize_t kds_custat_bin_show(struct file *filp, struct kobject *kobj,
+	struct bin_attribute *attr, char *buffer, loff_t offset, size_t count)
+{
+	struct xocl_dev *xdev = dev_get_drvdata(container_of(kobj, struct device, kobj));
+	ssize_t ret = 0;
+	u32 nread = 0;
+
+	/* Allocate a buffer as the buffer is too big to fit on the stack */
+	char *buf = kmalloc(MAX_CU_STAT_BUFFER_SIZE * sizeof(char), GFP_KERNEL);
+
+	/* Populate the allocated buffer with CU data */
+	mutex_lock(&xdev->dev_lock);
+	ret = show_kds_custat_raw(&XDEV(xdev)->kds, buf, MAX_CU_STAT_BUFFER_SIZE);
+	mutex_unlock(&xdev->dev_lock);
+
+	/* Set the amount to write out dependant on the number of bytes received */
+	ret = (ret < MAX_CU_STAT_BUFFER_SIZE) ? ret : MAX_CU_STAT_BUFFER_SIZE;
+
+	/* Validate the number of bytes to send out for the current operation */
+	if (count < ret - offset)
+		nread = count;
+	else
+		nread = ret - offset;
+
+	/* If no bytes are left to read finish the operation */
+	if (nread <= 0)
+		nread = 0;
+	else
+		memcpy(buffer, buf + offset, nread);
+
+	kfree(buf);
+	return nread;
+}
+
+static struct bin_attribute kds_custat_bin_attr = {
+	.attr = {
+		.name = "kds_custat_bin",
```

Comment:
Why adding a new sysfs node "kds_custat_bin" here? Should it still be kds_custat_raw?